### ISSUE TITLE
Input shape error using RegularGridInterpolator with "cubic" or "quintic"

### ISSUE THREAD

**Issue by cthenoz** [scipy/scipy#16905]
Hi everyone,

There seems to be a problem with the newly added interpolation methods "slinear", "cubic" and "quintic" for the `RegularGridInterpolator` class. Let's review a simple example reproducing the problem for better understanding. I have a 5x5 grid filled with random values along two dimensions. Initializing the interpolator works well. However when calling the interpolator function over `point`, I get different behaviours depending on the method selected:
* If `method='linear'` or `method='near'`, everything works fine, I get a result.
* If `method='cubic'` or `method='quintic'` I get a shape error `ValueError: Shapes of x (5,) and y (2, 5, 5) are incompatible`.
* If `method='slinear'` I get a type error `TypeError: only size-1 arrays can be converted to Python scalars`

```
from scipy.interpolate import RegularGridInterpolator
import numpy as np

x = np.arange(0, 5)
y = np.arange(0, 5)

data = np.random.randint(0, 100, (5, 5, 2))

interp = RegularGridInterpolator((x, y), data, method="cubic")

point = (0.5, 1.6)
out = interp(point)
print(out)

# 'linear'
# [71.4 65.9]

# 'nearest'
# [15. 24.]

# 'slinear'
# TypeError: only size-1 arrays can be converted to Python scalars

# 'cubic' or 'quintic'
# ValueError: Shapes of x (5,) and y (2, 5, 5) are incompatible
```

Am I getting something wrong?

**Comment by j-bowhay** [comment]
I believe this has recently been fixed in #16508 as the example works on the current main branch however I will let @ev-br confirm.

**Comment by ev-br** [comment]
Indeed, this has been fixed for scipy 1.10. Until it's released, can use e.g. nightly builds (with all the caveats of using nightly builds in production etc). Thanks for the report @cthenoz anyway, it's actually nice to see that this is being used!

**Comment by cthenoz** [comment]
Thanks for your replies! I note to use the nightly builds until the release of 1.10. Keep up with the great job ;-D!
